### PR TITLE
Add spider for Niteroi/RJ

### PR DESCRIPTION
**Pull request by mvinoba**
First contact with scrapy's spiders and writing a crawler because I want to contribute to this awesome project, reviews are very much appreciated! Niterói's page is just a simple form allowing to select a date and then redirects to a formatted URL, even when the gazette is not available. I ran the test suite but still not sure how the unavailable pages are to be handled.

### CODE REVIEW

File: processing/data_collection/gazette/spiders/rj_niteroi.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+from gazette.items import Gazette
+import datetime as dt
+import scrapy
+
+
+class RjNiteroiSpider(scrapy.Spider):
```
Comment (rennerocha) [inline processing/data_collection/gazette/spiders/rj_niteroi.py]: Please extend you spider from  `BaseGazetteSpider`
https://github.com/okfn-brasil/diario-oficial/blob/master/processing/data_collection/gazette/spiders/base.py.

File: processing/data_collection/gazette/spiders/rj_niteroi.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    month_names = ['01_Jan', '02_Fev', '03_Mar', '04_Abr', '05_Mai',
+                   '06_Jun', '07_Jul', '08_Ago', '09_Set', '10_Out', '11_Nov', '12_Dez']
+
+    def parse(self, response):
+        """
+        @url http://www.niteroi.rj.gov.br
+        @returns items 1
+        @scrapes date file_urls municipality_id power scraped_at
+        """
+        start_date = dt.date(2015, 1, 1)
```
Comment (rennerocha) [inline processing/data_collection/gazette/spiders/rj_niteroi.py]: I see no problem to get everything (not only from 2015-01-01). Items are dropped based on date in `BaseGazetteSpider` and there is nothing against getting everything available. According to Niteroi page, you can use 2003-07-01 as start date to retrieve the items:
"http://www.niteroi.rj.gov.br/do.html -  *Aviso: O diário oficial passou a ser fornecido a partir da data de Julho de 2003, o que inviabiliza qualquer D.O anterior."

File: processing/data_collection/gazette/spiders/rj_niteroi.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        parsing_date = dt.date.today()
+        while parsing_date >= start_date:
+            month = self.month_names[parsing_date.month-1]
```
Comment (rennerocha) [inline processing/data_collection/gazette/spiders/rj_niteroi.py]: `month = self.month_names[parsing_date.month - 1]`

**Comment by giovanisleite** [comment]
Hey, @mvinoba, update the [cities.md](https://github.com/okfn-brasil/diario-oficial/blob/master/CITIES.md)

**Comment by mvinoba** [comment]
@giovanisleite thanks for the ping! As a recent git user, since the PR was made before the creation of `cities.md`, does doing `git fetch upstream`, `git pull upstream master` and then editing, commiting and pushing the commits work fine? I made the update this way but I'm not sure if I should do this process by rebasing the branch instead.

Also, seems like #57 modifies the way URLs are shown in `cities.md`.

**Comment by giovanisleite** [comment]
@mvinoba you can put ```:done:``` there, when the pr be merged this crawler will be done

**Comment by mvinoba** [comment]
@giovanisleite fair point, done!

**Comment by jvanz** [comment]
What's the status of this PR? Is someone working on it? :)

**Comment by mvinoba** [comment]
> What's the status of this PR? Is someone working on it? :)

Hey @jvanz! If I recall correctly and things haven't changed this PR should be good to merge.

**Comment by rennerocha** [comment]
This PR was completed in https://github.com/okfn-brasil/querido-diario/pull/373/files

**Pull request closed without merging**